drivers/npmx/npmx_shell.c

```diff
@@ -2515,103 +2521,162 @@ static int cmd_gpio_debounce_set(const struct shell *shell, size_t argc, char **
 		shell_error(shell, "Error: shell is not initialized.");
 		return 0;
 	}
 
+	if (argc < 2) {
+		shell_error(shell, "Error: missing config value.");
+		return 0;
+	}
+
+	int err = 0;
+	uint32_t config_val = shell_strtoul(argv[1], 0, &err);
+	if (err != 0) {
+		shell_error(shell, "Error: config has to be an integer.");
+		return 0;
+	}
+
 	npmx_charger_t *charger_instance = npmx_charger_get(npmx_instance, 0);
 
 	uint32_t modules_mask;
 	npmx_error_t err_code = npmx_charger_module_get(charger_instance, &modules_mask);
-
 	if (!check_error_code(shell, err_code)) {
 		shell_error(shell, "Error: unable to get charger module status.");
 		return 0;
 	}
-
 	if ((modules_mask & NPMX_CHARGER_MODULE_CHARGER_MASK) != 0) {
 		shell_error(shell, "Error: charger must be disabled to set ADC NTC value.");
 		return 0;
 	}
 
-	npmx_adc_ntc_type_t type = (npmx_adc_ntc_type_t)data;
+	npmx_adc_t *adc_instance = npmx_adc_get(npmx_instance, 0);
+	npmx_adc_ntc_config_t ntc_config;
+	err_code = npmx_adc_ntc_config_get(adc_instance, &ntc_config);
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to read ADC NTC config.");
+		return 0;
+	}
 
-	if (type == NPMX_ADC_NTC_TYPE_HI_Z) {
-		err_code = npmx_charger_module_disable_set(charger_instance,
-							   NPMX_CHARGER_MODULE_NTC_LIMITS_MASK);
-		if (check_error_code(shell, err_code)) {
-			shell_info(
-				shell,
-				"Info: the NTC temperature limit control module has been disabled.");
-			shell_info(shell,
-				   "      To re-enable, change the NTC type to != ntc_hi_z.");
-		} else {
-			shell_error(
-				shell,
-				"Error: unable to disable the NTC temperature limit control module.");
+	switch (config_type) {
+	case ADC_NTC_CONFIG_TYPE_TYPE:
+		ntc_config.type = npmx_adc_ntc_type_convert(config_val);
+		if (ntc_config.type == NPMX_ADC_NTC_TYPE_INVALID) {
+			shell_error(shell, "Error: unable to convert resistance to NTC type.");
 			return 0;
 		}
-	} else {
-		err_code = npmx_charger_module_enable_set(charger_instance,
-							  NPMX_CHARGER_MODULE_NTC_LIMITS_MASK);
-		if (check_error_code(shell, err_code)) {
-			shell_info(
-				shell,
-				"Info: the NTC temperature limit control module has been enabled.");
+
+		if (ntc_config.type == NPMX_ADC_NTC_TYPE_HI_Z) {
+			err_code = npmx_charger_module_disable_set(
+				charger_instance, NPMX_CHARGER_MODULE_NTC_LIMITS_MASK);
+			if (check_error_code(shell, err_code)) {
+				shell_info(
+					shell,
+					"Info: the NTC temperature limit control module has been disabled.");
+				shell_info(shell,
+					   "      To re-enable, change the NTC type to != 0.");
+			} else {
+				shell_error(
+					shell,
+					"Error: unable to disable the NTC temperature limit control module.");
+				return 0;
+			}
 		} else {
-			shell_error(
-				shell,
-				"Error: unable to enable the NTC temperature limit control module.");
-			return 0;
+			err_code = npmx_charger_module_enable_set(
+				charger_instance, NPMX_CHARGER_MODULE_NTC_LIMITS_MASK);
```

Comment:
What if it's already enabled?
Same with already disabled.

Reply:
Can we change it in a different PR?

Reply:
It can be changed in shell refactor where a code is a little simpler.

Reply:
Added `if`.